Snake/BusinessLayer/maze.cs

```diff
@@ -1,137 +1,199 @@
-     public class maze
+    public class Maze
     {
-        private const int BORDER = 1;
-        private const int NO_BORDER = 0;
+        private const int border = 1;
+        private const int noBorder = 0;
+        private enum direction { horizon, vertical };
         private int width, height;
-        private int[,] Maze;
-        Random random_number;
+        private int[,] maze;
+        private Random randomNumber;
 
 
 
-        public maze(int w,int h)
+        public Maze(int w, int h)
         {
             width = w;
             height = h;
-            Maze = new int[height, width];
-            random_number = new Random();
+            maze = new int[height, width];
+            randomNumber = new Random();
         }
 
         public int[,] CreateMaze()
         {
-            Generate_Basic_Maze();
+            GenerateBorder();
+
+            int mazeMode = randomNumber.Next(1, 5);
+
+            switch (mazeMode)
+            {
+                case 1:
+                    GenerateGridMaze();
+                    break;
+
+                case 2:
+                    GenerateCrossMaze();
+                    break;
+
+                case 3:
+                    GenerateVerticalJungleMaze();
+                    break;
 
-            Generate_All_Obstacles();
-            
-            return Maze;
+                case 4:
+                    GenerateHorizonJungleMaze();
+                    break;
+
+                default:
+                    break;
+            }
+
+
+            return maze;
         }
 
 
-        public void Generate_Basic_Maze()
+        public void GenerateBorder()
         {
             for (int i = 0; i < height; i++)
             {
                 for (int j = 0; j < width; j++)
                 {
-                    Maze[i, j] = NO_BORDER;
+                    maze[i, j] = noBorder;
                 }
 
             }
 
             for (int i = 0; i < height; i++)
             {
-                //every colum of first row and last row should be marked as border 
                 if (i == 0 || i == height - 1)
                 {
                     for (int j = 0; j < width; j++)
                     {
-                        Maze[i, j] = BORDER;
+                        maze[i, j] = border;
                     }
                 }
 
-                //first and last colum of other rows should be marked as border
                 else
                 {
-                    Maze[i, 0] = BORDER;
-                    Maze[i, width - 1] = BORDER;
+                    maze[i, 0] = border;
+                    maze[i, width - 1] = border;
                 }
             }
 
         }
 
 
-
-        public void Generate_All_Obstacles()
+        public void GenerateGridMaze()
         {
-            int Vertical_Line_Head_X = random_number.Next(0,height);
-            int Vertical_Line_Head_Y = random_number.Next(width/2,width);
+            for (int i = 0; i < height - 2; i = i + 4)
```

Comment:
consider if you should explain where the magic number 4 comes from. i can not tell by reading the code. an easy way to get around this is create a constant or else add a comment